### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by apt2tell**
## Writing integration tests with Capybara gem and securing app from `n+1` problems.

> I and @Baayeh accomplished the following tasks in this features:

- Solved the `N+1` problem for the posts#index
- Implement test case for user show view
- Implement test case for post show view
- Add integration test case for User index page
- Add test case for User post index page

before:
post_query_before

after:
post_query_after

### CODE REVIEW

File: spec/integration/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      end
+    end
+
+    it "displays post's body" do
+      @posts.each do |post|
+        expect(page).to have_content(post.text)
+      end
+    end
+
+    it 'has first comments on a post' do
```
Comment (GzimAsani) [inline spec/integration/post_index_spec.rb:85]: - [ ] You have two failing tests kindly update them 
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello @Okechukwu-muokwugwo and @Baayeh                          :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with your testing, I really like how you tested them and you did 95% of the job, just few more things left +1: 

 ## Required Changes 
- [ ] Check the comments below :arrow_double_down: 
- _**If you are struggling with tests, use this [capybara cheat sheet](https://gist.github.com/zhengjia/428105)**_


 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/integration/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    it 'has a section pagination if there are more posts' do
+      expect(page).to have_content('Pagination')
+    end
+  end
+
+  it 'should redirect to the post details page' do
+    click_link('View More', href: user_post_path(@users.first, @posts.first))
+    expect(page).to have_current_path(user_post_path(@users.first, @posts.first))
+  end
+end
```
Comment (Whoistolu) [inline spec/integration/post_index_spec.rb]: - [ ] [REQUIRED] It looks like the previous reviewer missed out on this. Kindly note that  a part of the requirement states that you must have a featured test for `User` **post index page**:
- Tests that assert you can see how many likes a post has.

Kindly add this test as it is a core requirement.

File: spec/integration/post_index_spec.rb (hunk 1 above)
Comment (Whoistolu) [reply spec/integration/post_index_spec.rb:85]: - Well resolved @Okechukwu-muokwugwo 👍🏽

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @Okechukwu-muokwugwo 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by andersonlebon — APPROVED** [review]
## Hi @Okechukwu-muokwugwo  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**